python/pyarrow/tests/strategies.py

```diff
@@ -224,6 +231,9 @@ def _pymap(draw, key_type, value_type, size, nullable=True):
 
 @st.composite
 def arrays(draw, type, size=None, nullable=True):
+    pytest.importorskip("pytz")
```

Comment:
We might want to change this to use `zoneinfo` for recent Python versions, so we don't skip this full test just when pytz is not present.

Reply:
There are couple of issues I am tying to solve to make this work:
1. It seems all the tests in `test_strategies.py` are being skipped?
2. The code I need to change to use `zoneinfo` is here:
https://github.com/apache/arrow/blob/69682ec944d053a60316c0af903a167eca0f24db/python/pyarrow/tests/strategies.py#L263-L274

but: * I would need to install `timedelta,` which we do not depend on currently, to get the offset into a `timedelta` instance. ** The current code for the offset doesn't seem to work if I play with it locally. Example:

```python
>>> import pyarrow as pa
>>> 
>>> ty = pa.timestamp('s', tz='+07:30')
>>> pa.types.is_timestamp(ty)
True

>>> offset_hours = int(ty.tz) # This errors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '+07:30'

>>> import pytz
>>> tz = pytz.timezone(ty.tz) # This also errors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alenkafrim/repos/pyarrow-dev-9/lib/python3.9/site-packages/pytz/__init__.py", line 188, in timezone
    raise UnknownTimeZoneError(zone)
pytz.exceptions.UnknownTimeZoneError: '+07:30'
```
This works:
```python
>>> ty = pa.timestamp('s', tz='America/New_York')
>>> pa.types.is_timestamp(ty)
True
>>> pytz.timezone(ty.tz)
<DstTzInfo 'America/New_York' LMT-1 day, 19:04:00 STD>
```

My idea for a current change would be as follows. But it uses `timedelta` and the tests are being skipped so I am not sure if this works.
```python
  elif pa.types.is_timestamp(ty):
      min_int64 = -(2**63)
      max_int64 = 2**63 - 1
      min_datetime = datetime.datetime.fromtimestamp(min_int64 // 10**9)
      max_datetime = datetime.datetime.fromtimestamp(max_int64 // 10**9)
      try:
          offset = ty.tz.split(":")
          offset_hours = int(offset[0])
          offset_min = int(offset[1])
          tz = timedelta(hours=offset_hours, minutes=offset_min)
      except ValueError:
          tz = ZoneInfo(ty.tz)
      value = st.datetimes(timezones=st.just(tz), min_value=min_datetime,
                           max_value=max_datetime)
```

Reply:
> It seems all the tests in `test_strategies.py` are being skipped?

Yeah, it seems you need to manually enable them by passing `--enable-hypothesis` flag to pytest (I struggled with this in the past as well, I remember. This is documented .. but easy to miss: https://arrow.apache.org/docs/dev/developers/python.html#test-groups

> I would need to install `timedelta`,

That's not about the standard library `datetime.timedelta`?

Reply:
Thanks for the suggestion! I have totally missed the Test Groups part of the documentation.

As for the `timedelta` I got a bit confused, yes, you are absolutely right. Will add the change today.

Reply:
> 5 tests are still being skipped and I wasn't able to check the code change from the `test_arrays`

For seeing the reason something is skipped, you can pass `-r S` to pytest:

```
$ pytest python/pyarrow/tests/test_array.py --enable-hypothesis -r S
..
====================================== short test summary info ==============================
SKIPPED [1] python/pyarrow/tests/conftest.py:221: nopandas NOT enabled
SKIPPED [1] python/pyarrow/tests/strategies.py:236: could not import 'ZoneInfo': No module named 'ZoneInfo'
SKIPPED [2] python/pyarrow/tests/conftest.py:221: large_memory NOT enabled
====================================== 213 passed, 4 skipped in 0.87s =========================
```

So one test is skipped because of pandas being present, and two tests are skipped by default because they are "large memory" tests. One because `ZoneInfo` could not be imported, but so that was actually because of a bug here (the module name is lowercase, see the inline comment below).